### PR TITLE
CB-5734 Additional Validations for DistroX Autoscale API

### PR DESCRIPTION
**Pull request by smaniraju**
CB-5734 Additional Validations
1. Load Based Scaling Enabled only for clusters with cluster-proxy tunnel.
2. Additional Validation for invalid crn or name.

### CODE REVIEW

File: autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java [hunk 1]
```diff
@@ -229,14 +224,34 @@ private LoadAlertResponse createLoadAlertResponse(LoadAlert loadAlert) {
...
+        alertId.ifPresentOrElse(
+                updateAlert -> {
+                    validateAlertForUpdate(clusterId, updateAlert, AlertType.LOAD);
+                }, () -> {
+                    if (!cluster.getTunnel().useClusterProxy()) {
+                        throw new BadRequestException(String.format("Cluster '%s' is not configured with Cluster Proxy Tunnel, " +
+                                "Cluster Tunnel is %s. Load Based Scaling not supported", cluster.getStackCrn(), cluster.getTunnel()));
+                    }
+
+                    String hostGroup = json.getScalingPolicy().getHostGroup();
```
Comment (sidseth) [inline autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java:248]: Do we need to check for the existence of the hostGroup in the cluster?

File: autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java [hunk 2]
```diff
@@ -229,14 +224,34 @@ private LoadAlertResponse createLoadAlertResponse(LoadAlert loadAlert) {
...
 
-    private void validateLoadAlert(Long clusterId, Optional<Long> alertId, LoadAlertRequest json) throws ParseException {
-        validateAlertForUpdate(clusterId, alertId, AlertType.LOAD);
+    private void validateLoadAlert(Long clusterId, Optional<Long> alertId, LoadAlertRequest json) {
+        Cluster cluster = clusterService.findById(clusterId);
+        alertId.ifPresentOrElse(
+                updateAlert -> {
+                    validateAlertForUpdate(clusterId, updateAlert, AlertType.LOAD);
+                }, () -> {
+                    if (!cluster.getTunnel().useClusterProxy()) {
```
Comment (sidseth) [inline autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java:243]: This looks good based on the assumption that we're going with w.r.t all clusters using the ClusterProxy. Will have to include this in the documentation eventually.

File: autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java [hunk 3]
```diff
@@ -265,30 +280,27 @@ private PrometheusAlertResponse createPrometheusAlertResponse(PrometheusAlert al
...
-                    alert = null;
-
-            }
-            if (alert == null) {
-                throw new NotFoundException(String.format("Could not find %s alert with id: '%s', for cluster: '%s'", alertType, alertId, clusterCrn));
-            }
-        });
+    private void validateAlertForUpdate(Long clusterId, Long alertId, AlertType alertType) {
+        BaseAlert alert;
+        String clusterCrn = clusterService.findStackCrnById(clusterId);
```
Comment (sidseth) [inline autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java:285]: Nit: Move into the (alert == null) block.

Comment (smaniraju) [reply autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java:285]: ack

File: autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java (hunk 1 above)
Comment (smaniraju) [reply autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java:248]: we would also need the hostGroup services validation, which can be added based on our other discussion.

**Pull request merged**